File: 217-ContainsDuplicate/217-ContainsDuplicate.java
// Last updated: 9/14/2025, 4:59:06 AM
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> res= new HashSet<>();

        for(int num: nums){
            if(res.contains(num)){
                return true;
            }
            res.add(num);
        }return false;
    }
}

File: 217-ContainsDuplicate/217-ContainsDuplicate.js
// Last updated: 3/1/2026, 11:44:38 PM
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    const result = new Set();
    const seen = new Set();

    for ( let x of nums){
        if (seen.has(x)){
            return true
        }else{
            seen.add(x)
        }
    }return false
};